### PR TITLE
Bash runner

### PR DESCRIPTION
**Pull request by Ellmen**
This time the bash script is ~~almost~~ actually working!
Edit: working now

### CODE REVIEW

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+threads=2
+pam_start=0
+pam_end=1
+range=$[$pam_end-$pam_start+1]
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: change $[ ... ] to $(( ... ))

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+results_dir="results"
+
+# run docking scripts for ranges in parallel
+echo "Launching scripts"
+for i in $( seq 0 $[$threads-1] ); do
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: change $[ ... ] to $(( ... ))

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	if [ $i = $[$threads-1] ]; then
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: change $[ ... ] to $(( ... ))

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		python dock_variants.py -s=$[$range/$threads*$i+$pam_start] -e=$pam_end
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: change $[ ... ] to $(( ... ))

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	else
+		python dock_variants.py -s=$[$range/$threads*$i+$pam_start] -e=$[$range/$threads*($i+1)+$pam_start-1] &
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: change $[ ... ] to $(( ... ))

File: models/tridimensional/docking_validation/dock_caller.sh (hunk 1 above)
Comment (Ellmen) [inline models/tridimensional/docking_validation/dock_caller.sh]: are those different?

**Comment by jrhawley** [comment]
So I changed around a lot of your dock_caller.sh script. Not sure where I should put it, so I'll put the link to the [gist I made](https://gist.github.com/jrhawley/cc1a9a17d0dadfef088a)

**Comment by Ellmen** [comment]
That version didn't generate complete ranges when I ran it (missing 5-7, 13-15, 21-23, 29-31). Also, the sleep 1 line was because the scripts were running so simultaneously that they were both trying to generate a results directory at the same time which was crashing it. Another thing is it's running the CSV script before the docking script finishes executing.

I know mine's kind of messy but I promise it works :P

**Comment by jrhawley** [comment]
Yeah I changed a couple things after I posted it :p Look at it now, it should be OK now? not sure

**Comment by Ellmen** [comment]
Didn't quite work, I played around with it a little to get the ranges working. Also, it balances the remainder better now.

**Comment by jrhawley** [comment]
Yeah, I wasn't entirely comfortable with the way it looked yesterday. Just ran a couple tests, dock_caller.sh seems to work well

**Comment by mattsmart** [comment]
So its good? what about forced argument passing for threads and label?

File: models/tridimensional/docking_validation/dock_caller.sh [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+step=$(($range/$threads))
+remainder=$(($range % $threads))
+pids=()
+
+# amount of remainder to distribute to tasks
+remainder_plus=$(($remainder/$threads+1))
+
+# set so pam_first is initially pam_start
+pam_last=$((pam_start-1))
```
Comment (jrhawley) [inline models/tridimensional/docking_validation/dock_caller.sh]: not sure if this needs to be $pam_start

Comment (Ellmen) [inline models/tridimensional/docking_validation/dock_caller.sh]: hm I think it should be, it worked but I'll change it

**Comment by mattsmart** [comment]
closing bc unnecessary files, gonna repost PR

**Pull request closed without merging**